src/Http/Http.Abstractions/test/EndpointHttpContextExtensionsTests.cs

```diff
@@ -65,7 +65,9 @@ public void SetEndpoint_NullOnContextWithoutFeature_NoFeatureSet()
         context.SetEndpoint(null);
 
         // Assert
-        Assert.Null(context.Features.Get<IEndpointFeature>());
+        var endpointFeature = context.Features.Get<IEndpointFeature>();
```

Comment:
Would it be worthwhile to confirm that `context.Endpoint` agrees?  Would it be worthwhile to try the same test with `context.Endpoint = null`, rather than `SetEndpoint`?